bin/varnishncsa/varnishncsa.c

```diff
@@ -809,10 +814,22 @@ dispatch_f(struct VSL_data *vsl, struct VSL_transaction * const pt[],
 				e--;
 
 			switch (tag) {
+				/* The tests "if(!CTX.b_opt) break;" and
+				   "if (CTX.b_opt && tag == ...) break;"
+				   are here for your safety. We are in
+				   backend mode if and only if t->type ==
+				   VSL_t_bereq, and fields like ReqMethod
+				   should never occur in backend
+				   mode. However, assert() is too strong,
+				   so we just ignore such "impossible"
+				   cases. */
 			case SLT_HttpGarbage:
 				skip = 1;
 				break;
 			case SLT_PipeAcct:
+				/* PipeAcct is only possible for requests,
+				   but to be safe we check: */
+				if (CTX.b_opt) break;
```

Comment:
This is correct, but I prefered this over other options I could think of. The choice was between
* Making two switch statements, one for each mode. This would result in code duplication, which is very bad.
* Doing asserts. In this case, varnishncsa would die if, for example, we saw a SLT_PipeAcct when in backend mode. This is probably not desired behavior.
* The current solutions, where "impossible" fields are ignored, and as much code as possible is shared between the two modes. In my opinion, this is the shortest path to dual mode, if this is ever going to be implemented.

I may have overlooked a better solution, of course.